src/guardian/keys-validation/keys-validation.service.ts

```diff
@@ -129,8 +116,8 @@ export class KeysValidationService {
   }
 
   private async updateCache(validatedKeys: [Key & DepositData, boolean][]) {
-    validatedKeys.forEach(([key, isValid]) =>
-      this.keysCache.set(key.key, { signature: key.depositSignature, isValid }),
+    validatedKeys.forEach(([depositData, isValid]) =>
+      this.depositDataCache.set(JSON.stringify(depositData), isValid),
```

Comment:
I would like to point out that `JSON.stringify` does not handle `Buffer` correctly. In this specific code, it doesn't seem to be a problem, but as a good practice, it would be better to explicitly handle fields with `Buffer`.

```ts
  private serializeDepositData(depositData: DepositData): string {
    return JSON.stringify({
      ...depositData,
      withdrawalCredentials: depositData.withdrawalCredentials.toString('hex'),
      genesisForkVersion: depositData.genesisForkVersion.toString('hex'),
    });
  }
```